odoo/sql_db.py

```diff
@@ -248,6 +248,7 @@ def __init__(self, pool, dbname, dsn, **kwargs):
         # default log level determined at cursor creation, could be
         # overridden later for debugging purposes
         self.sql_log_count = 0
+        self.sql_track_table_count = False
```

Comment:
A public flag on the cursor is crappy, and your code updating it is incorrect.
Please make the flag private, and add a context manager to update it.
```py
@contextmanager
def _enable_table_tracking(self):
    try:
        old = self._sql_table_tracking
        self._sql_table_tracking = True
        yield
    finally:
        self._sql_table_tracking = old
```